src/python/common/actions.py

```diff
@@ -378,23 +379,25 @@ def execute(self, conf=None, caller=None, apMon=None, logger=None):
                             syncFlag=False,
                             logger=logger)
         calledLimit = False
-        customclassID = int(self.options['port']) - 50000
         try:
             try:
-                #######
                 try:
-                    if 'hostDest' in self.options and 'port' in self.options:
+                    if 'rate' in self.options and 'clientIP' in self.options and \
+                       'hostDest' in self.options and 'port' in self.options:
                         logger.info("Calling TC command with %s" % self.options)
-                        # limit add destPort rate sourceIP destIP port
-                        exitCode = os.system("limit add %s %s %s %s %s" % (customclassID, 500,
+                        # limit add <class> <rate> <sourceIP> <destIP> <destPort>
+
+                        # Use monID as a hash of tc class
+                        monID = hashlib.sha224(self.options['monID'], "%s:%s" % (self.options['hostDest'], self.options['port'])).hexdigest()
```

Comment:
Please use if 'monID' in self.options:
    ............
else:
    monID = randominteger....